browsers/duckduckgo.safariextension/js/background.js

```diff
@@ -160,28 +161,23 @@ var onBeforeNavigation = function (e) {
     
     // skip upgrading broken sites:
     if (thisTab.site.isBroken) {
-        console.log('HTTPS: temporarily skip upgrades for: ' + e.url)
+        console.log('HTTPS: temporarily skip upgrades for: ' + url)
         return
     }
 
-    // avoid redirect loops:
-    // TODO: this won't work in safari yet:
-    //if (thisTab.httpsRedirects[e.url] >= 7) {
-    //    console.log('HTTPS: cancel https upgrade. redirect limit exceeded for url: \n' + e.url)
-    //    e.preventDefault()
-    //    e.target.url = thisTab.downgradeHttpsUpgradeRequest(e)
-    //}
+    // skip trying again if we've already tried upgrading this url
+    if (thisTab.hasUpgradedUrlAlready(url)) {
+        console.log('HTTPS: skipping upgrade to avoid redirect loops', url)
+        return
+    }
 
-    const url = e.url
-    const isMainFrame = true // always main frame in this handler
-    const upgradedUrl = https.getUpgradedUrl(e.url, thisTab, isMainFrame)
+    const upgradedUrl = https.getUpgradedUrl(url, thisTab, isMainFrame)
 
     if (url.toLowerCase() !== upgradedUrl.toLowerCase()) {
         console.log('HTTPS: upgrade request url to ' + upgradedUrl)
         thisTab.upgradedHttps = true
-        thisTab.addHttpsUpgradeRequest(upgradedUrl)
+        thisTab.addHttpsUpgradeRequest(upgradedUrl, url)
 
-        e.preventDefault()
```

Comment:
removing `e.preventDefault()` seems to have fixed autocomplete from redirecting on upgrade to a page that's pre-fetching and everything seems to still work fine without it as far as I can tell..